src/litdata/utilities/env.py

```diff
@@ -49,8 +49,14 @@ def detect(cls) -> "_DistributedEnv":
             world_size = torch.distributed.get_world_size()
             global_rank = torch.distributed.get_rank()
             # Note: On multi node CPU, the number of nodes won't be correct.
-            num_nodes = world_size // torch.cuda.device_count() if torch.cuda.is_available() else world_size
-            if torch.cuda.is_available() and world_size % torch.cuda.device_count() != 0:
+            if torch.cuda.is_available() and world_size // torch.cuda.device_count() >= 1:
+                num_nodes = world_size // torch.cuda.device_count()
+            else:
+                num_nodes = 1
+
+            # If you are using multiple nodes, we assume you are using all the GPUs.
+            # On single node, a user can be using only a few GPUs of the node.
+            if torch.cuda.is_available() and num_nodes >= 1 and world_size % torch.cuda.device_count() != 0:
```

Comment:
I think this line would still fail, the issue is `world_size % torch.device_count` is not necessarily 0

Reply:
This is gated by `num_nodes >= 1`. So this is checked only for multi nodes.

Reply:
Yes, good catch ! Can you push it ?